rheem-platforms/rheem-giraph/src/main/java/org/qcri/rheem/giraph/operators/GiraphPageRankOperator.java

```diff
@@ -0,0 +1,258 @@
+
+    private Tuple<Collection<ExecutionLineageNode>, Collection<ChannelInstance>> runGiraph(
+            FileChannel.Instance inputFileChannelInstance,
+            StreamChannel.Instance outputChannelInstance,
+            GiraphExecutor giraphExecutor,
+            OptimizationContext.OperatorContext operatorContext)
+            throws IOException, URISyntaxException, ClassNotFoundException, InterruptedException {
+
+        assert inputFileChannelInstance.wasProduced();
+        Configuration configuration = operatorContext.getOptimizationContext().getConfiguration();
+        String tempDirPath = this.getPathOut(configuration);
+
+        PageRankParameters.setParameter(PageRankParameters.PageRankEnum.ITERATION, this.getNumIterations());
+
+
+        FileSystem fs = FileSystems.getFileSystem(tempDirPath).orElseThrow(
+                () -> new RheemException(String.format("Cannot access file system of %s.", tempDirPath))
+        );
+
+        fs.delete(tempDirPath, true);
+
+        final String inputPath = inputFileChannelInstance.getSinglePath();
+
+        GiraphConfiguration conf = giraphExecutor.getGiraphConfiguration();
+        //vertex reader
+        conf.set("giraph.vertex.input.dir", inputPath);
+        conf.set("giraph.maxWorkers", "1");
+        conf.set("giraph.SplitMasterWorker", "false");
+        conf.set("mapreduce.output.fileoutputformat.outputdir", tempDirPath);
+       // conf.set("mapreduce.output.basename", "rheem");
+        conf.setComputationClass(PageRankAlgorithm.class);
+        conf.setVertexInputFormatClass(
+                PageRankAlgorithm.PageRankVertexInputFormat.class);
+        conf.setWorkerContextClass(
+                PageRankAlgorithm.PageRankWorkerContext.class);
+        conf.setMasterComputeClass(
+                PageRankAlgorithm.PageRankMasterCompute.class);
+        conf.setNumComputeThreads(1);
+
+        conf.setVertexOutputFormatClass(PageRankAlgorithm.PageRankVertexOutputFormat.class);
+
+        GiraphJob job = new GiraphJob(conf, "rheem-giraph");
+        job.run(true);
+
+        final String actualInputPath = FileSystems.findActualSingleInputPath(tempDirPath);
+        Stream<Tuple2<Long, Float>> stream = this.createStream(actualInputPath);
+
+        outputChannelInstance.accept(stream);
+
+        final ExecutionLineageNode mainExecutionLineage = new ExecutionLineageNode(operatorContext);
+        mainExecutionLineage.add(LoadProfileEstimators.createFromSpecification(
+                "rheem.giraph.pagerank.load.main", configuration
+        ));
+        mainExecutionLineage.addPredecessor(inputFileChannelInstance.getLineage());
+
+        final ExecutionLineageNode outputExecutionLineage = new ExecutionLineageNode(operatorContext);
+        outputExecutionLineage.add(LoadProfileEstimators.createFromSpecification(
+                "rheem.giraph.pagerank.load.output", configuration
+        ));
+        outputChannelInstance.getLineage().addPredecessor(outputExecutionLineage);
```

Comment:
I think, we only need one load estimator for this operator. We should split into two estimators if they correspond to different partial operations of the operator and we can measure the runtime of those operations independently. I think, that is not the case here because. Unlike GraphChi, when Giraph is finished, the data already resides in the HDFS, ready to read without further transformation. I am not 100% sure, though.

Compare, for instance, with `JavaMapOperator`. I think, we can go here with `ExecutionOperator.modelEagerExecution`.